WordPress/Classes/Services/MenusService.m

```diff
@@ -85,83 +85,39 @@ - (void)syncMenusForBlog:(Blog *)blog
 
 #pragma mark - Creating and updating menus
 
-- (void)createMenuWithName:(NSString *)menuName
-                      blog:(Blog *)blog
-                   success:(nullable MenusServiceCreateMenuRequestSuccessBlock)success
+- (void)createOrUpdateMenu:(Menu *)menu
+                   forBlog:(Blog *)blog
+                   success:(nullable MenusServiceCreateOrUpdateMenuRequestSuccessBlock)success
                    failure:(nullable MenusServiceFailureBlock)failure
 {
     NSParameterAssert([blog isKindOfClass:[Blog class]]);
-    NSAssert([self blogSupportsMenusCustomization:blog], @"Do not call this method on unsupported blogs, check with blogSupportsMenusCustomization first.");
-    
-    if (blog.wordPressComRestApi == nil) {
-        return;
-    }
-
-    MenusServiceRemote *remote = [[MenusServiceRemote alloc] initWithWordPressComRestApi:blog.wordPressComRestApi];
-    [remote createMenuWithName:menuName
-                          blog:blog
-                       success:^(RemoteMenu * _Nonnull remoteMenu) {
-                           [self.managedObjectContext performBlock:^{
-                               if (success) {
-                                   success(remoteMenu.menuID);
-                               }
-                           }];
-                       }
-                       failure:failure];
-}
-
-- (void)updateMenu:(Menu *)menu
-           forBlog:(Blog *)blog
-           success:(nullable MenusServiceUpdateMenuRequestSuccessBlock)success
-           failure:(nullable MenusServiceFailureBlock)failure
-{
-    NSParameterAssert([blog isKindOfClass:[Blog class]]);
     NSParameterAssert([menu isKindOfClass:[Menu class]]);
     NSAssert([self blogSupportsMenusCustomization:blog], @"Do not call this method on unsupported blogs, check with blogSupportsMenusCustomization first.");
     
     if (blog.wordPressComRestApi == nil) {
-        return;
+        return;   
```

Comment:
This is a bit of a `?` actually. Looking at https://github.com/wordpress-mobile/WordPress-iOS/commit/c090e31736e0fdd6b5be74ca7aece38eb1868d3b, these were added to prevent creating a remote with a `nil` api object.

@koke can you shed some light on this one since you wrote the commit? Since Menus relies on a `wordPressComRestApi` maybe an assertion would be the best choice.